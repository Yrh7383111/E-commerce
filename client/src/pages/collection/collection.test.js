import React from 'react';
import Router from 'react-router-dom';
import * as redux from 'react-redux';
import { shallow } from 'enzyme';
import CollectionPage from './collection.component';
import CollectionItem from '../../components/collection-item/collection-item.component';


jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: jest.fn(),
}));

describe('CollectionPage component', () => {
    let wrapper;
    let useParams;
    let useSelector;

    const mockItems = [{ id: 1 }, { id: 2 }, { id: 3 }];
    const mockCollection = {
        title: 'Test',
        items: mockItems
    };
    const mockCollectionId = {
        collectionId: 1
    };


    beforeEach(() => {
        useParams = jest.spyOn(Router, 'useParams').mockReturnValue(mockCollectionId);
        useSelector = jest.spyOn(redux, 'useSelector').mockReturnValue(mockCollection);

        wrapper = shallow(<CollectionPage />);
    });

    afterEach(() => {
        useParams.mockClear();
        useSelector.mockClear();
    });

    test('should render the CollectionPage component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should render the same number of CollectionItems as mockItems', () => {
        expect(wrapper.find(CollectionItem).length).toBe(mockItems.length);
    });
});