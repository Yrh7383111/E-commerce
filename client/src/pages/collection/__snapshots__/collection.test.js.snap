// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPage component should render the CollectionPage component 1`] = `ShallowWrapper {}`;
